runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcCommitWorkStream.java

```diff
@@ -305,7 +297,7 @@ long getBytes() {
     }
   }
 
-  private class Batcher {
+  private class Batcher implements CommitWorkStream.RequestBatcher {
 
     private final Map<Long, PendingRequest> queue;
```

Comment:
Wouldn't this still throw `ConcurrentModificationException` if `queue` is accessed by multiple threads? Should we make this `ConcurrentHashMap` vs `HashMap`?

Reply:
See other comment, batcher is not thread-safe but we shouldn't have a batcher member just vend new batchers. Was leftover from testing.